include/cudaq/Optimizer/Dialect/Quake/QuakeTypes.td

```diff
@@ -50,6 +50,16 @@ def WireType : QuakeType<"Wire", "wire"> {
   let genStorageClass = 0;
 }
 
+def QControlType : QuakeType<"QControl", "qcontrol"> {
```

Comment:
What do we do when we have a ssa value of type qcontrol that then gets used on a subsequent op as a target?

Reply:
The basics of the design are that a "normal" wire can be converted to a control wire with the `convert_ctrl` op. A control wire value would be strongly type checked such that it could only be used in control argument positions and nowhere else.

The incoming "normal" wire would have a 1+1 use rule due to the volatile nature of wire values. A "normal" wire can only be used once as a target and have a +1 possible use if input to a `convert_ctrl` op. We can verify those constraints quite easily. There are also global constraints to check when verifying the function. The `convert_ctrl` must dominate the normal target use, and all uses of that `convert_ctrl` must likewise dominate the normal target use. If those constraints are not met, the IR is in an invalid state.

While another type isn't strictly needed here, I think it has some utility in making the IR a bit easier to read than without it. The `convert_ctrl` op is mostly a NOP, but it does make it clear and obvious when and where a wire will be used as a control qubit.